### DISCUSSION TITLE
question on algorithm

### DISCUSSION THREAD

**Discussion by Doubl3MD** [halide/Halide#7051] [Q&A]
Hi,

first off: Halide is amazing! 
But also complex if one is rather new to the knowledge domain. :)

Can somebody give me a hint on how to achieve the following type of pipeline in a feed forward fashion.
The attached image is for clarification of what I want to achieve.

The inputs are like this: We have an input image/buffer (8x8 example). We also have a tile that needs to be rendered a times b proportion. And the output ( x*y) depending on an arbitrary scale.

The algorithm should go like this: 

For every pixel of the input image, render a render tile on the output multiplied by the color of the input pixel. Note that on the output the rendertile is bigger then the fraction of 'output width/input width(8)' thus resulting in overlapping of every next tile rendered.

If this overlap is not present it would be an easy one, but now I can not say output(x,y) = some expression with color value. Because the color could change depending on the next tile by tile rendering as explained.

Any hint would be great! :)

Thx!

![image](https://user-images.githubusercontent.com/113417608/192243772-362e4c7e-55f3-4eae-b01e-0c77e0bdcc56.png)

**Comment by ashishUthama** [comment]
Not sure if I understood correctly, but maybe this:

```
output(x,y) = 0.0f;
output(x+rx,y+ry) = output(x,y) + input(inx,iny)*renderTiel(rx,ry)

```

You would have to replace the '+' with whatever logic you need to combine/accumulate the renderings.

**Reply by Doubl3MD** [reply]
Thx for the answer! I'm getting up to speed with halide and I'm getting used to the feed forward mindset.
